src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.ReadHelper.cs

```diff
@@ -97,18 +106,22 @@ public partial class JsonTypeInfo<T>
         internal sealed override object? DeserializeAsObject(Stream utf8Json)
             => Deserialize(utf8Json);
 
-        internal T? ContinueDeserialize(
+        internal bool ContinueDeserialize(
             ref ReadBufferState bufferState,
             ref JsonReaderState jsonReaderState,
-            ref ReadStack readStack)
+            ref ReadStack readStack,
+            out T? value)
```

Comment:
For reference type Ts, I'm curious if you measured any worst case impact to these changes that changed it to returning the T to out'ing the T?

Reply:
I saw no meaningful changes in performance when running the microbenchmark suite against the PR branch.